task/discover_granules_s3.py

```diff
@@ -133,6 +149,14 @@ def discover(self, response_iterator):
                         # )
                         pass
 
+                # Check Time
+                time_remaining = self.lambda_context.get_remaining_time_in_millis()
+                if time_remaining < 30000:
```

Comment:
If the lambda's timeout was set < 30 seconds, would this create an infinite loop? Can we add a "just in case" catch here that makes sure that the lambda timeout > 30 sec since that's a variable we're allowing to be passed into GDG?

Reply:
It is not possible for this to cause an infinite loop. If the timeout <= 30 seconds then it would discover only 1 key per execution of the lambda. 
I will however just make the early return threshold configurable.